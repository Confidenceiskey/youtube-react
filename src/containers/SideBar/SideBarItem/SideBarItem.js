import React, { Component } from 'react';
import { Icon, Menu } from 'semantic-ui-react';
import { Link, withRouter } from 'react-router-dom';
import './SideBarItem.scss';

class SideBarItem extends Component {

  shouldBeHighLighted = () => {
    const { pathname } = this.props.location;
    const path = this.props.path;
    console.log('pathname = ', pathname);

    if (path === '/') {
      return pathname === path;
    }
    return pathname.includes(path);
  }

  render() {
    const { icon, label, path } = this.props;
    // React will ignore custom boolean attributes, therefore we pass a string
    // we use this attribute in our SCSS styling
    const highLight = this.shouldBeHighLighted() ? 'highlight-item' : null;
    return (
      <Link to={{pathname: path}}>
        <Menu.Item className={['sidebar-item', highLight].join(' ')}>
          <div className='sidebar-item-alignment-container'>
            <span>
              <Icon size='large' name={icon} />
            </span>
            <span>{label}</span>
          </div>
        </Menu.Item>
      </Link>
    );
  }
}

export default withRouter(SideBarItem);
