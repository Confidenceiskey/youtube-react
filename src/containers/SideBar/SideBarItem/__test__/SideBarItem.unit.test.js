import React from 'react';
import { shallow } from 'enzyme';
import SideBarItem from '../SideBarItem';

describe('Various cases for SideBarItem', () => {
  it('renders empty SideBarItem', () => {
    const wrapper = (
      <SideBarItem />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders highlighted SideBarItem', () => {
    const wrapper = (
      <SideBarItem highlighted icon='fire' label='Trending' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders SideBarItem w/ icon & label', () => {
    const wrapper = (
      <SideBarItem icon='fire' label='Trending' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
