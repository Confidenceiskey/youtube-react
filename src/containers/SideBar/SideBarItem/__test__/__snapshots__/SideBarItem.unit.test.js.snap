// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Various cases for SideBarItem renders SideBarItem 1`] = `<SideBarItem />`;

exports[`Various cases for SideBarItem renders SideBarItem w/ icon & label 1`] = `
<SideBarItem
  icon="fire"
  label="Trending"
/>
`;

exports[`Various cases for SideBarItem renders empty SideBarItem 1`] = `<SideBarItem />`;

exports[`Various cases for SideBarItem renders highlighted SideBarItem 1`] = `
<SideBarItem
  highlighted={true}
  icon="fire"
  label="Trending"
/>
`;
